v2/extensions/dataref_extension.go

```diff
@@ -0,0 +1,36 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package extensions
+
+import (
+	"github.com/cloudevents/sdk-go/v2/event"
+	"net/url"
+)
+
+const DataRefExtensionKey = "dataref"
+
+type DataRefExtension struct {
+	DataRef string `json:"dataref"`
```

Comment:
Not that it matters much but I'm curious why it's a struct instead of just a string? Are you leaving the option open that one day we may need to include extra metadata along with the URL?